riposte-core/src/test/java/com/nike/riposte/server/handler/RequestInfoSetterHandlerTest.java

```diff
@@ -93,4 +121,114 @@ public void doChannelRead_checks_for_fully_send_responses_but_does_nothing_else_
         assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
     }
 
+    @Test
+    public void doChannelRead_returns_do_not_fire_when_null_state() {
+        // given
+        doReturn(true).when(stateMock).isResponseSendingLastChunkSent();
+
+        // when
+        PipelineContinuationBehavior result = handler.doChannelRead(ctxMock, httpContent);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.DO_NOT_FIRE_CONTINUE_EVENT);
+    }
+
+    @Test
+    public void doChannelRead_throws_exception_when_no_request_info_when_HttpContent_message() {
+        // given
+        doReturn(null).when(stateMock).getRequestInfo();
+        Exception thrownException = null;
+
+        // when
+        try {
+            handler.doChannelRead(ctxMock, httpContent);
```

Comment:
AssertJ has a convenience method for all this:

```
Throwable ex = Assertions.catchThrowable(() -> handler.doChannelRead(ctxMock, httpContent));
```

Then you'd assert something on the exception which implicitly verifies the exception was thrown since if the assertion was executed on null it would also fail. The existing assertion in this test is sufficient, for example.